doc/source/data/batch_inference.rst

```diff
@@ -0,0 +1,752 @@
+            :end-before: __hf_quickstart_load_end__
+
+    .. group-tab:: PyTorch
+
+        For the PyTorch example of this guide, we'll create a NumPy array with 100
+        entries, which represents the input to a feed-forward neural network.
+
+        .. literalinclude:: ./doc_code/pytorch_quick_start.py
+            :language: python
+            :start-after: __pt_quickstart_load_start__
+            :end-before: __pt_quickstart_load_end__
+
+    .. group-tab:: TensorFlow
+
+        For the TensorFlow example of this guide, we'll create a NumPy array with 100
+        entries, which represents the input to a feed-forward neural network.
+
+        .. literalinclude:: ./doc_code/tf_quick_start.py
+            :language: python
+            :start-after: __tf_quickstart_load_start__
+            :end-before: __tf_quickstart_load_end__
+
+2. Setting up your Model
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Next, you want to set up your model for inference, by defining a predictor.
+The core idea is to define a class that loads your model in its ``__init__`` method and
+and implements a ``__call__`` method that takes a batch of data and returns a batch of predictions.
+Below you find examples for PyTorch, TensorFlow, and HuggingFace.
+
+.. tabs::
+
+    .. group-tab:: HuggingFace
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/hf_quick_start.py
+                :language: python
+                :start-after: __hf_quickstart_model_start__
+                :end-before: __hf_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+    .. group-tab:: PyTorch
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/pytorch_quick_start.py
+                :language: python
+                :start-after: __pt_quickstart_model_start__
+                :end-before: __pt_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+
+    .. group-tab:: TensorFlow
+
+        .. callout::
+
+            .. literalinclude:: ./doc_code/tf_quick_start.py
+                :language: python
+                :start-after: __tf_quickstart_model_start__
+                :end-before: __tf_quickstart_model_end__
+
+            .. annotations::
+                <1> Use the constructor (``__init__``) to initialize your model.
+
+                <2> The ``__call__`` method runs inference on a batch of data.
+
+
+3. Getting Predictions with Ray Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once you have your Ray Datastream ``ds`` and your predictor class, you can use
+``ds.map_batches(...)`` to get predictions.
+Mapping batches this way is the recommended way to run inference with Ray, but there are also other options.
```

Comment:
```suggestion
Mapping batches is the recommended way to run inference with Ray, but there are also other options.
```

Reply:
You mention this in the callout below. New users only need to know about map_batches.